controllers/k8ssandra/vector.go

```diff
@@ -57,3 +58,21 @@ func (r *K8ssandraClusterReconciler) reconcileVector(
 	dcLogger.Info("Vector Agent ConfigMap successfully reconciled")
 	return result.Continue()
 }
+
+// setupVectorCleanup adds owner references to ensure that the remote resources created by reconcileVector are correctly
+// cleaned up when the CassandraDatacenter is deleted. We do that in a second pass because the CassandraDatacenter did
+// not exist yet at the time those resources were created.
+func (r *K8ssandraClusterReconciler) setupVectorCleanup(
+	ctx context.Context,
+	kc *k8ssandraapi.K8ssandraCluster,
```

Comment:
Suggestion: with these functions, it would be nice not to pass the whole chunk of `kc` state, and just pass a string instead. Since I think we're just grabbing the kc.SanitizedName() in all cases.